### ISSUE TITLE
Can't upload to imgur

### ISSUE THREAD

**Issue by kohheepeace** [NickeManarin/ScreenToGif#903]
Hi, I can't upload gif images to imgur. **version: 2.30**, download from installer.

This is a error log.
![image](https://user-images.githubusercontent.com/29557494/119274366-803cc480-bc4a-11eb-98c6-9bc2e8ce9974.png)

The gif file is less than 10MB, so I think this issue is not related with imgur api limit.

I can upload gif files to imgur by manually drug and drop.

Thanks for this awesome software.

----

*I have confirmed that I can upload in the old **version 2.27.3**.

**Comment by pawlos** [comment]
@NickeManarin  I did a brief look and I can see that 2.27.3 had correct Imgur Client-ID and secret in the app, that's no longer the case. But not sure how it supposed to work in the current app. I see no way to set those via UI for anonymous mode. How it supposed to work now?

**Comment by NickeManarin** [comment]
@pawlos Version 2.30 has the Client-ID and secrets normally as version 2.27.3 has.

What changed is that the user needs to create a profile for uploading gifs to Imgur.

------------

This issue may be related to the fact that the app reached the Imgur limit:

```
{"data":{"error":"Too Many Requests","request":"\/oauth2\/authorize","method":"POST"},"success":false,"status":429}
```

**Comment by pawlos** [comment]
@NickeManarin oh, ok. I was checking in VS and there was no info about those keys. I guess they are added to the released version. I was getting a 400 - invalid request, as there was no `CliendID` set.

**Comment by the-m4a** [comment]
I was seeing the same issue, even after authenticating.  The issue went away once I pointed it at an existing album, for whatever that's worth.
I am using whatever the latest version available is (2.30... I updated ScreenToGif immediately before capturing and uploading).  I could not find any log files - the Logs folder under my Program Files (x86)\ScreenToGif contained only something from 5/12.  Are the logs in a different place or is there something to do in order to enable them?

**Comment by pawlos** [comment]
@NickeManarin Checked with 2.30.1 (Imgur anonymous) and I'm getting a invalid Client-ID anyway, but yeah the Client-ID and secret are the same as in 2.27.3. Also tried that version (2.27.3) but upload didn't work for me too (also giving me 403).

![image](https://user-images.githubusercontent.com/1296768/119931898-93b99980-bf82-11eb-8bb9-a5edb98fe3cb.png)

**Comment by NickeManarin** [comment]
Ok thanks, I'm going to check that.

**Comment by NickeManarin** [comment]
Issue fixed.

When sending images as anonymous, a unnecessary content-type was being added to the multipart data:

```cs
stream.WriteStringUtf8("Content-Type: text/plain; charset=utf-8");
```

**Comment by NickeManarin** [comment]
> I was seeing the same issue, even after authenticating. The issue went away once I pointed it at an existing album, for whatever that's worth.
> I am using whatever the latest version available is (2.30... I updated ScreenToGif immediately before capturing and uploading). I could not find any log files - the Logs folder under my Program Files (x86)\ScreenToGif contained only something from 5/12. Are the logs in a different place or is there something to do in order to enable them?

Same issue as above.

I also improved the logging, so it should create logs with the response when something goes wrong.
